Framework/PythonInterface/plugins/algorithms/SCDAdjustSampleNorm.py

```diff
@@ -0,0 +1,246 @@
+        if len(vanfile) > 0 and not van_ws.isDefault:
+            issues['VanadiumWorkspace'] = "Cannot specify both a vanadium file and workspace"
+
+        # Verify given workspaces exist
+        if not input_ws.isDefault:
+            input_ws_list = input_ws.value.split(",")
+            for ws in input_ws_list:
+                if not mtd.doesExist(ws):
+                    issues['InputWorkspaces'] = "Could not find input workspace '{}'".format(ws)
+
+        return issues
+
+    def PyExec(self):
+        load_van = not self.getProperty("VanadiumFile").isDefault
+        load_files = not self.getProperty("Filename").isDefault
+
+        if load_files:
+            datafiles = self.getProperty("Filename").value
+        else:
+            datafiles = self.getProperty("InputWorkspaces").value.split(",")
+
+        vanadiumfile = self.getProperty("VanadiumFile").value
+        vanws = self.getProperty("VanadiumWorkspace").value
+        height = self.getProperty("DetectorHeightOffset").value
+        distance = self.getProperty("DetectorDistanceOffset").value
+        method = self.getProperty("OutputAsMDEventWorkspace").value
+
+        if method:
+            minvals = self.getProperty("MinValues").value
+            maxvals = self.getProperty("MaxValues").value
+            merge = self.getProperty("MergeInputs").value
+            wslist = []
+        else:
+            bin0 = self.getProperty("BinningDim0").value
+            bin1 = self.getProperty("BinningDim1").value
+            bin2 = self.getProperty("BinningDim2").value
+
+        out_ws = self.getPropertyValue("OutputWorkspace")
+        out_ws_name = out_ws
+
+        if load_van:
+            vanws = Load(vanadiumfile, StoreInADS=False)
+
+        has_multiple = True if len(datafiles) > 1 else False
+
+        # Default wavelength in WANDSCDtoQ if not set in the input file
+        wavelength = 1.488
+
+        for file in datafiles:
+            if load_files:
+                scan = Load(file, LoadHistory=False, StoreInADS=False)
+            else:
+                scan = mtd[file]
+
+            self.log().information("Processing file '{}'".format(file))
+
+            # If processing multiple files, append the base name to the given output name
+            if has_multiple:
+                if load_files:
+                    out_ws_name = out_ws + "_" + os.path.basename(file).strip(',.nxs')
+                else:
+                    out_ws_name = out_ws + "_" + file
+
+            exp_info = scan.getExperimentInfo(0)
+
+            # Adjust detector height and distance with new offsets
+            if height != 0.0 or distance != 0.0:
+                # Move all the instrument banks (MoveInstrumentComponents does not work on MDHisto Workspaces)
+                component = exp_info.componentInfo()
+                for bank in range(1, 4):
+                    index = component.indexOfAny("bank{}".format(bank))
+
+                    offset = V3D(0.0, height, distance)
```

Comment:
I'm still not sure that doing the detector distance offset that way is correct. The height is fine.

I think you need to find the current position of the detector in the x-z plane and move it in that direction away from 0. This should probably be clarified with the CIS/IS.